content/blog/2024/01/25/2024-01-25-jenkins-2023-recap.adoc

```diff
@@ -0,0 +1,231 @@
+With dozens of CVEs published every month, we prefer not to publish information for each one unless it is relevant.
+
+To reduce the occurrence of those issues and provide reporters with a point of reference, we have updated the link:https://github.com/jenkinsci/docker/blob/master/SECURITY.md[security policy in Docker].
+
+We would also like to extend our gratitude to the link:https://securitylab.github.com/[GitHub Security Lab] for reporting 27 vulnerabilities through their link:https://codeql.github.com/[CodeQL] rules and for their collaborative efforts in both the analysis and the enhancement of our tools.
+
+The team does not rely solely on “external” security researchers to find vulnerabilities; a considerable amount of time during the year is spent on proactively auditing the code.
+This year was marked by some specific research areas: OAuth flow and disabled SSL validation, in addition to the usual CSRF, XSS, or XXE vulnerabilities.
+
+Additionally, we have expanded the scope of our security reviews for Pull Requests, particularly in Jenkins core.
+Approximately link:https://github.com/jenkinsci/jenkins/pulls?q=is%3Apr+label%3Asecurity-approved[70 PRs] were labeled as “security-approved”.
+This proactive measure allowed the team to prevent the introduction of some vulnerabilities, which is significantly less costly than addressing them after they have been incorporated into the production code.
+
+As part of our mission, raising community awareness about security is also a key factor in reducing the introduction of vulnerabilities at an early stage.
+In this regard, we welcomed Andrea to our security team for a summer internship from June to August.
+By the end of it, he had successfully reported 28 vulnerabilities.
+For more information about his internship, you can read link:/blog/2023/08/23/summer-internship-in-jenkins-security/[his blog post].
+
+Education and tooling are closely related for us.
+We are investing time to propose new security automation and to improve our existing tooling.
+This year, we link:https://groups.google.com/g/jenkinsci-dev/c/-wTosY82jZU/m/8OidfVTcAAAJ[introduced] the existing CodeQL ruleset for hosting requests.
+This provided future maintainers with detailed explanations of the findings.
+Along with this effort, we also enhanced our security scan by link:https://github.com/jenkins-infra/jenkins-security-scan/pull/13[supporting warning suppression] through code annotation.
+
+Continuing with tooling improvements, we link:https://groups.google.com/g/jenkinsci-dev/c/8Ia0OBmzF1A/m/YBdi40u3BAAJ[added] a beta feature to the bot present in SECURITY tickets, allowing maintainers to block or unblock releases of their plugins, to reduce the likelihood of an involuntary release when one is staged in private for a future advisory.
```

Comment:
```suggestion
Continuing with tooling improvements, we link:https://groups.google.com/g/jenkinsci-dev/c/8Ia0OBmzF1A/m/YBdi40u3BAAJ[added] a beta feature to the bot present in SECURITY tickets, allowing maintainers to block or unblock releases of their plugins.
The intent of this feature is to reduce the likelihood of an involuntary release when one is staged in private for a future advisory.
```
Suggestion to separate the two sentences, and add a bit of syntax so that everything connects and makes sense.